orchestra/model/configuration.py

```diff
@@ -367,32 +367,31 @@ def _get_branches(self):
             assert type(branch) is str, "branches must be a list of strings"
         return branches
 
-
-def hash_config_dir(config_dir):
-    hash_script = f"""find "{config_dir}" -type f -print0 | sort -z | xargs -0 sha1sum | sha1sum"""
-    config_hash = subprocess.check_output(hash_script, shell=True).decode("utf-8").strip().partition(" ")[0]
-    return config_hash
-
-
-def run_ytt(orchestra_dotdir, use_cache=True):
-    config_dir = os.path.join(orchestra_dotdir, "config")
-    config_cache_file = os.path.join(orchestra_dotdir, "config_cache.yml")
-    config_hash = hash_config_dir(config_dir)
-
-    if use_cache and os.path.exists(config_cache_file):
-        with open(config_cache_file, "r") as f:
-            cached_hash = f.readline().replace("#!", "").strip()
-            if config_hash == cached_hash:
-                return f.read()
-
-    ytt = os.path.join(os.path.dirname(__file__), "..", "support", "ytt")
-    env = os.environ.copy()
-    env["GOCG"] = "off"
-    expanded_yaml = subprocess.check_output(f"'{ytt}' -f {config_dir}", shell=True, env=env).decode("utf-8")
-
-    if use_cache:
-        with open(config_cache_file, "w") as f:
-            f.write(f"#! {config_hash}\n")
-            f.write(expanded_yaml)
-
-    return expanded_yaml
+    def load_configuration(self, use_cache=True):
+        config_dir = os.path.join(self.orchestra_dotdir, "config")
+        config_cache_file = os.path.join(self.orchestra_dotdir, "config_cache.json")
+        config_hash = self.hash_config_dir()
+
+        if use_cache and os.path.exists(config_cache_file):
+            with open(config_cache_file) as f:
+                cached_config = json.load(f)
+                if config_hash == cached_config.get("config_hash"):
+                    return cached_config["config"], cached_config["yaml_config"]
+
+        ytt = os.path.join(os.path.dirname(__file__), "..", "support", "ytt")
+        env = os.environ.copy()
+        env["GOCG"] = "off"
+        expanded_yaml = subprocess.check_output(f"'{ytt}' -f {config_dir}", shell=True, env=env).decode("utf-8")
+        parsed_config = yaml.safe_load(expanded_yaml)
+
+        if use_cache:
+            with open(config_cache_file, "w") as f:
+                json.dump({"config_hash": config_hash, "config": parsed_config, "yaml_config": expanded_yaml}, f)
```

Comment:
Didn't we say to create two separate files?

Reply:
I misunderstood and thought you were ok-ing to save the yaml version inline.
FTR there's basically no measurable overhead with doing this and a positive effect is that the two versions can't go out of sync, also less confusion by the user re: which one is used.
But I can use two files, up to you.

Reply:
Having a separate `.yml` file is much more readable.
`less .orchestra/*yml`